site/src/pages/WorkspacesPage/data.ts

```diff
@@ -0,0 +1,113 @@
+import {
+  QueryKey,
+  useMutation,
+  useQuery,
+  useQueryClient,
+} from "@tanstack/react-query"
+import { getWorkspaces, updateWorkspaceVersion } from "api/api"
+import {
+  Workspace,
+  WorkspaceBuild,
+  WorkspacesResponse,
+} from "api/typesGenerated"
+
+const getQueryKey = (page: number, query: string) => {
+  return ["workspaces", query, page]
+}
```

Comment:
Is `getQueryKey` potentially going to be used elsewhere?  I am curious why it is a separate function.

Also just to make sure I understand, passing the `query` and `page` as part of the query key makes it run `getWorkspaces` again if those parameters change, right?

Reply:
> Is getQueryKey potentially going to be used elsewhere? I am curious why it is a separate function.

It was, but it is not anymore. I'm going to remove this function.

> Also just to make sure I understand, passing the query and page as part of the query key makes it run getWorkspaces again if those parameters change, right?

Exactly!